src/api/git/reset.int.test.ts

```diff
@@ -0,0 +1,185 @@
+    const initialCommit = await git.commit({
+      fs,
+      dir,
+      message: 'initial commit',
+      author,
+    });
+
+    await fs.promises.writeFile(newFileName, '');
+    await git.add({ fs, dir, filepath: newFileName });
+    await git.commit({
+      fs,
+      dir,
+      message: 'new commit',
+      author,
+    });
+
+    await reset({
+      dir,
+      branch: 'master',
+      commitId: initialCommit,
+      hard: true,
+    });
+
+    const commits = await git.log({
+      fs,
+      dir,
+      depth: 1,
+    });
+
+    const directory = await fs.promises.readdir(dir);
+
+    expect(commits[0].oid).toEqual(initialCommit);
+    expect(directory).not.toContain(newFileName);
+  });
+
+  it('should keep new files which are not committed in the working directory', async () => {
+    const initialCommit = await git.commit({
+      fs,
+      dir,
+      message: 'initial commit',
+      author,
+    });
+
+    await fs.promises.writeFile(newFileName, '');
+
+    await reset({
+      dir,
+      branch: 'master',
+      commitId: initialCommit,
+      hard: true,
+    });
+
+    const commits = await git.log({
+      fs,
+      dir,
+      depth: 1,
+    });
+
+    const directory = await fs.promises.readdir(dir);
+
+    expect(commits[0].oid).toEqual(initialCommit);
+    expect(directory).toContain(newFileName);
+  });
+
+  it('should remove files which are staged', async () => {
+    const initialCommit = await git.commit({
+      fs,
+      dir,
+      message: 'initial commit',
+      author,
+    });
+
+    await fs.promises.writeFile(newFileName, '');
+    await git.add({ fs, dir, filepath: newFileName });
+
+    await reset({
+      dir,
+      branch: 'master',
+      commitId: initialCommit,
+      hard: true,
+    });
+
+    const commits = await git.log({
+      fs,
+      dir,
+      depth: 1,
+    });
+
+    const directory = await fs.promises.readdir(dir);
+
+    expect(commits[0].oid).toEqual(initialCommit);
+    expect(directory).not.toContain(newFileName);
+  });
+
+  it('should revert files which were modified', async () => {
+    await fs.promises.writeFile(newFileName, 'hello');
+    await git.add({ fs, dir, filepath: newFileName });
+    const initialCommit = await git.commit({
+      fs,
+      dir,
+      message: 'initial commit',
+      author,
+    });
+
+    await fs.promises.writeFile(newFileName, 'hello world');
+    await git.add({ fs, dir, filepath: newFileName });
+    await git.commit({
+      fs,
+      dir,
+      message: 'new commit',
+      author,
+    });
+
+    await reset({
+      dir,
+      branch: 'master',
+      commitId: initialCommit,
+      hard: true,
+    });
+
+    const commits = await git.log({
+      fs,
+      dir,
+      depth: 1,
+    });
+
+    const file = await fs.promises.readFile(newFileName);
+
+    expect(commits[0].oid).toEqual(initialCommit);
+    expect(Buffer.from(file).toString()).toBe('hello');
```

Comment:
You can do `readFile(newFileName, 'utf8')` to get a string back and avoid the manual buffering